test/cases/cross-distro.sh

```diff
@@ -75,17 +75,79 @@ if ! sudo curl -s --unix-socket $APISOCKET http:///localhost/api/status > /dev/n
     exit 1
 fi
 
-if ! DISTROS=$(sudo curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
+if ! RECOGNIZED_DISTROS=$(sudo curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
     echo "osbuild-composer server error getting distros list. 😢"
     exit 1
 fi
 
-REMAINDER=$(echo "$DISTROS" | jq -r '.distros' | grep -v -E "$PATTERN")
-if [ -n "$REMAINDER" ]; then
+# check if the list of installed repositories by the RPM matches reality
+INSTALLED_DISTROS=$(find "/usr/share/osbuild-composer/repositories" -name '*.json' | sort)
+if [ "$INSTALLED_DISTROS" == "$DISTROS_FROM_RPM" ]; then
+    echo "All distros from RPM are in the expected location."
+else
+    echo "Unexpected or missing distros detected!"
+    echo "Expected distros:"
+    echo "$DISTROS_FROM_RPM"
+    echo "Actual distros:"
+    echo "$INSTALLED_DISTROS"
+    echo "Diff INSTALLED/FROM_RPM:"
+    diff <(echo "${INSTALLED_DISTROS}") <(echo "${DISTROS_FROM_RPM}")
+    exit 1
+fi
+
+# check if only expected repositories are recognized
+RECOGNIZED_REMAINDER=$(echo "$RECOGNIZED_DISTROS" | jq -r '.distros' | grep -v -E "$PATTERN")
+if [ -n "$RECOGNIZED_REMAINDER" ]; then
     echo "🔥 Unexpected distros installed:"
-    echo "$REMAINDER"
+    echo "$RECOGNIZED_REMAINDER"
     exit 1
 else
-    echo "🎉 All tests passed."
-    exit 0
+    echo "Only expected distros are installed."
+fi
+
+# clone osbuild-composer repository if not in it already
+if [ ! -d "repositories/" ]; then
+    git clone --depth 1 http://github.com/osbuild/osbuild-composer
+    REPO_PATH="osbuild-composer/repositories/"
+else
+    REPO_PATH="repositories/"
 fi
+
+# get all possible distros and leave only the unsupported ones
+ALL_DISTROS=$(find "$REPO_PATH" -name '*.json' -printf '%P\n' | awk -F "." '{ print $1 }')
+ALL_REMAINDERS=$(echo "$ALL_DISTROS" | grep -v -E "$PATTERN")
+
+# Push a blueprint with unsupported distro to see if composer fails gracefuly
+for REMAINING_DISTRO in $ALL_REMAINDERS; do
+    TEST_BP=blueprint.toml
+    tee "$TEST_BP" > /dev/null << EOF
+name = "bash"
+description = "A base system with bash"
+version = "0.0.1"
+distro= "$REMAINING_DISTRO"
+
+[[packages]]
+name = "bash"
+EOF
+    RESPONSE=$(sudo composer-cli blueprints push $TEST_BP 2>&1)
+
+    # there is a different reponse if legacy composer-cli is used
+    if rpm -q --quiet weldr-client; then
+        EXPECTED_RESPONSE="ERROR: BlueprintsError: '$REMAINING_DISTRO' is not a valid distribution"
+    else
+        EXPECTED_RESPONSE="'$REMAINING_DISTRO' is not a valid distribution"
+        RESPONSE=${RESPONSE#*: }
+    fi
+
+    if [ "$RESPONSE" == "$EXPECTED_RESPONSE" ];then
+            echo "Blueprint push with $REMAINING_DISTRO distro failed succesfuly"
```

Comment:
`failed sucesfully` -> `failed as expected` b/c a succesfull fail is an oxymoron.